filament/backend/src/opengl/PlatformCocoaGL.mm

```diff
@@ -38,8 +39,9 @@
     NSOpenGLContext* mGLContext = nullptr;
     NSView* mCurrentView = nullptr;
     std::vector<NSView*> mHeadlessSwapChains;
-    NSRect mPreviousBounds = {};
-    void updateOpenGLContext(NSView *nsView, bool resetView);
+    std::map<NSView*, NSRect> mPreviousSwapChainBounds;
```

Comment:
1. use unordered_map instead
2. we in fact don't need a map at all. But the implementation of SwapChain needs to change. Currently, we're simply casting the nsView to a Platform::SwapChain -- instead, we need to allocate a small CocoaGLSwapChain that stores the nsView and the bounds, and return that (and of course destroy that small structure when the swapchain is destroyed)

Reply:
Agreed with @pixelflinger.  @prewettg can you make these changes?

Reply:
@prideout Sure, will do. Do you have a preference on which of the options to go with?  Option 2, changing SwapChain, seems like it would simplify things a lot, so I'm inclined to go with that one, but can do whichever works best for the project.

Reply:
Thanks!  We definitely want option 2.  :)

Reply:
Implemented option 2. I found some additional problems with our (Open3D) case, when resizing. The resizing problems don't seem to happen with the SDL sample I added, it may be related to our use of GLFW instead of SDL.